Tests/RxSwiftTests/DisposeBagTest.swift

```diff
@@ -73,7 +72,32 @@ extension DisposeBagTest {
         XCTAssert(disposable1.count == 1)
         XCTAssert(disposable2.count == 1)
     }
+}
+
+// DisposeBag function builder initializer tests
+extension DisposeBagTest {
+  func testDisposableBuilder() {
+    let disposable1 = TestDisposable()
+    let disposable2 = TestDisposable()
+    let disposable3 = TestDisposable()
+
+    var disposeBag = DisposeBag {
```

Comment:
@freak4pc Just an idea that this might open the possibility of having a syntax like this for `zip`, `combineLastest` ... and so on? 
Maybe don't add much value, but it could feel nice to have something like
```swift
let observable = Zip {
   observable1
   observable2
}
```

Reply:
Interesting in general. I think @MortyMerr's idea of leveraging an ArrayBuilder would be good because it would mean having a possibility to have this for many things. I'll make some attempts :)

Reply:
What about it? Can I open separate PR for it with adding function builder overload for all array accepting methods?

Reply:
No need as I already have a PR for dispose bag.

Reply:
I mean, what about other operators which accepts array as a parameter (like zip, merge)? Will it be ok to add function builder overload for them?

Reply:
Let's think about this separately. After re-thinking this yesterday I'm wondering if it's really useful, since most use cases don't really do so much multi-line, where for a `DisposeBag` probably all cases do. I guess we _could_ provide both options, but I'm not sure it would provide any specific value.

Reply:
I thought about this some more — I don't want to over-abuse this for things like operators (because it's too wide). 

For DisposeBag I think it's mostly OK since it's just syntactic, there are no multiple types in the array (only `Disposable`) so the tweet @MortyMerr mentioned is not an issue really cc/ @LucianoPAlmeida